code/espurna/sensors/BME680Sensor.h

```diff
@@ -0,0 +1,287 @@
+
+        // Descriptive name of the sensor
+        String description() {
+            char buffer[21];
+            snprintf(buffer, sizeof(buffer), "BME680 @ I2C (0x%02X)", _address);
+            return String(buffer);
+        }
+
+        // Type for slot # index
+        unsigned char type(unsigned char index) {
+            if (index == 0) return MAGNITUDE_TEMPERATURE;
+            if (index == 1) return MAGNITUDE_HUMIDITY;
+            if (index == 2) return MAGNITUDE_PRESSURE;
+            if (index == 3) return MAGNITUDE_RESISTANCE;
+            if (index == 4) return MAGNITUDE_IAQ_ACCURACY;
+            if (index == 5) return MAGNITUDE_IAQ_STATIC;
+            if (index == 6) return MAGNITUDE_IAQ;
+            if (index == 7) return MAGNITUDE_CO2;
+            if (index == 8) return MAGNITUDE_VOC;
+
+            return MAGNITUDE_NONE;
+        }
+
+        // The maximum allowed time between two `bsec_sensor_control` calls depends on
+        // configuration profile `bsec_config_iaq` below. With these settings, the
+        // maximum allowed time is 3s.
+        void tick() {
+            _error = SENSOR_ERROR_OK;
+
+            if (!_isSensorOk()) {
+              _error = SENSOR_ERROR_OTHER;
+              return;
+            }
+
+            if (iaqSensor.run()) {
+              _rawTemperature = iaqSensor.rawTemperature;
+              _rawHumidity = iaqSensor.rawHumidity;
+              _temperature = iaqSensor.temperature;
+              _humidity = iaqSensor.humidity;
+              _pressure = iaqSensor.pressure / 100;
+              _gasResistance = iaqSensor.gasResistance;
+              _iaqAccuracy = iaqSensor.iaqAccuracy;
+              _iaq = iaqSensor.iaq;
+              _iaqStatic = iaqSensor.staticIaq;
+              _co2Equivalent = iaqSensor.co2Equivalent;
+              _breathVocEquivalent = iaqSensor.breathVocEquivalent;
+            }
+        }
+
+        // Current value for slot # index
+        double value(unsigned char index) {
+            if (index == 0) return _temperature;
+            if (index == 1) return _humidity;
+            if (index == 2) return _pressure;
+            if (index == 3) return _gasResistance;
+            if (index == 4) return _iaqAccuracy;
+            if (index == 5) return _iaq;
+            if (index == 6) return _iaqStatic;
+            if (index == 7) return _co2Equivalent;
+            if (index == 8) return _breathVocEquivalent;
+            return 0;
+        }
+
+    protected:
+
+        bool _isSensorOk(void) {
+          if (iaqSensor.status != BSEC_OK) {
+            #if SENSOR_DEBUG
+              DEBUG_MSG("[BME680] BSEC error code %s", iaqSensor.status);
+            #endif
+            return false;
+          }
+
+          if (iaqSensor.bme680Status != BME680_OK) {
+            #if SENSOR_DEBUG
+              DEBUG_MSG("[BME680] Error code %s", iaqSensor.bme680Status);
+            #endif
+            return false;
+          }
+
+          return true;
```

Comment:
tbh I'd probably use default false. We just had the exact same issue in init code, where default true fallen through i2c detection check.